### PR TITLE
feat: unify header structure

### PR DESCRIPTION
**Pull request by pregno**
* Add Header classes on `components.css`
* Unify Header structure between HTML and React
* Adopt same classes
* Add documentation page

### CODE REVIEW

File: apps/docs/content/3-components/2-library/16-header.mdx [hunk 1]
```diff
@@ -11,6 +11,283 @@ The GOV.IE header component tells users they’re using a service on GOV.IE and
...
+
+<Tabs id="tab2">
+  <TabList>
+    <TabItem value="tab21" checked>HTML</TabItem>
+    <TabItem value="tab22">Macro</TabItem>
+    <TabItem value="tab23">React</TabItem>
+  </TabList>
+  <TabPanel value="tab21">        
+    ```html
+
```
Comment (hamza14khan) [inline apps/docs/content/3-components/2-library/16-header.mdx:114]: question: Is it suppose to be empty?

**Review by hamza14khan — APPROVED** [review]
One minor comment other than that looks good.

File: apps/docs/content/3-components/2-library/16-header.mdx (hunk 1 above)
Comment (pregno) [reply apps/docs/content/3-components/2-library/16-header.mdx:114]: Nope, thanks

**Comment by BogdanDanilescu** [comment]
noticed a few discrepancies while comparing the React Header with the HTML Header
- for the `With No Search` story the divider is visible in HTML while in React it is not
- for the tablet view while opening the side menu, we don't have the layout effect anymore while in React we have it

**Review by BogdanDanilescu — APPROVED** [review]
LGTM

**Pull request merged**